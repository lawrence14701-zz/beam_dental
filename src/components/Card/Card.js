import Box  from '@material-ui/core/Box';
import MuiCard  from '@material-ui/core/Card';
import React from 'react';
import useStyles from './style';


const Card = (props) => {
    const {children} = props
    const classes = useStyles()
    return (
        <MuiCard className={classes.root}>
            <Box component="div" m={1}>{children}</Box>
        </MuiCard>
    )
}

export default Card