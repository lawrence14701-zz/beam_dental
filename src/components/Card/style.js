import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles(() => ({
  root: {
    padding: "12px",
    width: "100%",
    margin: "20px",
  },
}))

export default useStyles
