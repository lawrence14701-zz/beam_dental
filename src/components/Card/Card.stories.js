import React from "react"
import Card from "./Card"
import { Blue } from "../IconAndText/IconAndText.stories"
import { ThemeProvider } from "@material-ui/styles"
import muiTheme from "../../Theme/theme"
import Grid from "@material-ui/core/Grid"

export default {
  title: "Card",
  component: Card,
  decorators: [
    (Story) => (
      <ThemeProvider theme={muiTheme}>
        <Story />
      </ThemeProvider>
    ),
  ],
}

const Template = (args) => (
  <div>
    <Grid container="flex" spacing={8}>
      <Grid item>
        <Card {...args} />
      </Grid>
      <Grid item>
        <Card {...args} />
      </Grid>
    </Grid>
    <Grid container="flex" spacing={8}>
      <Grid item>
        <Card {...args} />
      </Grid>
      <Grid item>
        <Card {...args} />
      </Grid>
    </Grid>
  </div>
)

export const Primary = Template.bind({})

Primary.args = {
  children: <Blue color="blue" text="brush" num="2" />,
}
