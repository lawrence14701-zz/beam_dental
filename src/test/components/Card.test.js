import React from "react"
import Card from "../../components/Card/Card"
import "../jestsetup"

function createTestProps() {
  return {
    color: "blue",
    text: "brush",
    num: "2",
  }
}

test("render a list of items", () => {
  const props = createTestProps()
  const wrapper = shallow(<Card {...props} />)
  expect(wrapper).toMatchSnapshot()
})
