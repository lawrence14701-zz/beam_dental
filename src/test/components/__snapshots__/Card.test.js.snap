// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a list of items 1`] = `ShallowWrapper {}`;
